plugins/quay/src/api/index.ts

```diff
@@ -10,7 +10,7 @@ import {
   SecurityDetailsResponse,
 } from '../types';
 
-const DEFAULT_PROXY_PATH = '/quay/api';
+const DEFAULT_PROXY_PATH = '/quay';
```

Comment:
Why is this being changed? What is the benefit? This change is not reflected in the readme therefore users expose `/quay/api` in their proxy settings and are confused.

https://github.com/janus-idp/backstage-plugins/tree/main/plugins/quay#getting-started

Reply:
This change was made because the test `QuayApiClient should use a correct default proxy path` was failing if the default proxy path was `/quay/api` since the `/api` part is already inserted in the api functions like [`getTags`](https://github.com/janus-idp/backstage-plugins/blob/main/plugins/quay/src/api/index.ts#L93).

So I think in a real deployment the functions are using paths with `*/api/api/*` inside the encoded url but this works because the quay api sends a redirect see this call for example:
```
curl -v -H "X-Requested-With: XMLHttpRquest" https://quay.io/api/api/v1/repository/janus-idp/redhat-backstage-build/manifest/sha256:e461dc54b4e2469bb7f5bf85a4b7445c175548ba9d56c3f617dd25bc3adf3752/security
```
and the fetcher handles the redirect. However the mocked quay API doesn't send a redirect and that is why the test fails while using the `/quay/api` as the default proxy path. 

I see two ways how to approach this:
- we can either change the documentation so that the `API` in the config is supposed to only be `/quay` and the default should also be `/quay` 
- or we can remove the `/api` from the API function calls in `index.ts` and let the default be `/quay/api` and don't change the documentation

What do you think is the correct approach?

Reply:
I think your problem is a product of you locally setting the target to `https://quay.io/api` instead of `https://quay.io`. Not a product of setting the proxy path.

with default proxy path `/quay/api` and target `https://quay.io` the proxy resolves:

`/quay/api/api/v1/repository/janus-idp/redhat-backstage-build...`

gets translated to:

`https://quay.io/api/v1/repository/janus-idp/redhat-backstage-build...`

this is the correct resolution.

Reply:
That's true, I totally forgot about the proxy. I will try to also mock the proxy translation in the tests.

Reply:
I reverted the change to the default proxy path.